R/ClassifTask.R

```diff
@@ -56,10 +57,10 @@ makeClassifTaskDesc = function(id, data, target, weights, blocking, positive, sp
 #' @export
 print.ClassifTask = function(x, ...) {
   # remove 1st newline
-  di = printToChar(table(getTaskTargets(x)), collapse = NULL)[-1L]
+  di = printToChar(x$task.desc$class.distribution)
   m = length(x$task.desc$class.levels)
   print.SupervisedTask(x)
   catf("Classes: %i", m)
-  catf(collapse(di, "\n"))
+  catf(di)
```

Comment:
1) I'm not sure about this, but it looks to me like you're not removing the first newline any more (as the comment in line 59 says). I guess that's ok, but maybe in return you have to add `newline = FALSE` to the `catf("Classes: %i", m)` before this.
2) This was already a bug before: apparently this `catf` will fail if the number of classes is very large:
```R
> makeClassifTask("test", data.frame(x = 1:1000, y = as.factor(1:1000)), "y")
Supervised task: test
Type: classif
Target: y
Observations: 1000
Features:
   numerics     factors     ordered functionals
          1           0           0           0
Missings: FALSE
Has weights: FALSE
Has blocking: FALSE
Classes: 1000
Error in sprintf(...) : 'fmt' length exceeds maximal format length 8192
```
(The `sprintf(...)` referenced is exactly the `catf(di)`.) Can you try to use `cat(di)` instead, and then add a `"\n"` to the next line (`catf("\nPositive class....`)?

Reply:
P.S. that `catf` is broken in more ways than just one:
```R
> makeClassifTask("test", data.frame(x = 1:3, y = as.factor(c('a', 'b', '%s'))), "y")
Supervised task: test
[...]
Error in sprintf(...) : too few arguments
```

Reply:
Thank you for reviewing. I implemented your suggested changes.

`makeClassifTask("test", data.frame(x = 1:1000, y = as.factor(1:1000)), "y")`
and `makeClassifTask("test", data.frame(x = 1:3, y = as.factor(c('a', 'b', '%s'))), "y")`
do not throw an error anymore.